src/components/slider/index.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useState } from 'react';
+import {
+  Slider as ChakraSlider,
+  SliderMark as ChakraSliderMark,
+  SliderFilledTrack,
+  SliderThumb,
+  SliderTrack,
+} from '@chakra-ui/react';
+
+type SliderMark<T> = {
+  stepValue?: number;
+  label: string;
+  value: T;
+};
+
+type SliderProps<T> = {
+  useIndentation?: boolean;
+  marks: SliderMark<T>[];
+  defaultValue: SliderMark<T>;
```

Comment:
This prop is a bit misleading as the name implies we should provide a value but actually it expect a mark. Can we improve this by either providing a mark value or maybe renaming the prop?